packages/webclient/src/AuthHandler/AzCliAuthHandler.ts

```diff
@@ -1,77 +1,100 @@
+            await this._handler.initialize();
         }
-        return this.endpoint;
+
+        return this._handler;
     }
 
     public get subscriptionID(): string {
-        return this._subscriptionID;
+        return this._subscriptionId;
     }
 
     public get baseUrl(): string {
-        return this._baseUrl;
+        return this._cloudEndpoints['resourceManager'] || 'https://management.azure.com/';
+    }
+
+    public getCloudSuffixUrl(suffixName: string): string {
+        return this._cloudSuffixes[suffixName];
+    }
+
+    public getCloudEndpointUrl(endpointName: string): string {
+        return this._cloudEndpoints[endpointName];
+    }
+
+    public async initialize() {
+        let azAccountDetails = JSON.parse(await this.executeAzCliCommand('account show'));
+        let azCloudDetails = JSON.parse(await this.executeAzCliCommand('cloud show'));
+
+        this._subscriptionId = azAccountDetails['id'];
+        this._cloudSuffixes = azCloudDetails['suffixes'];
+        this._cloudEndpoints = azCloudDetails['endpoints'];
     }
 
     public async getToken(force?: boolean, args?: string[]): Promise<string> {
         if(!this._token || force) {  
             try {
-                let azAccessToken = JSON.parse(await executeAzCliCommand('account get-access-token', !!args ? args : []));
+                let azAccessToken = JSON.parse(await this.executeAzCliCommand('account get-access-token', !!args ? args : []));
+                console.log(`::add-mask::${azAccessToken}`);
                 this._token = azAccessToken['accessToken'];
             }
             catch(error) {
                 console.log('Failed to fetch Azure access token');
                 throw error;
             }          
         }
+
         return this._token;
     }
-}
+    
+    public static async getAzCliPath(): Promise<string> {
+        if (!this._azCliPath) {
+            this._azCliPath = await io.which('az', true);
+        }
+
+        return this._azCliPath;
+    }
 
-export async function initialize() {  
-    azPath = await io.which("az", true);
-    azCloudDetails = JSON.parse(await executeAzCliCommand('cloud show'));
-}
+    public async executeAzCliCommand(command: string, args?: string[]): Promise<string> {
+        let azCliPath = await AzCliAuthHandler.getAzCliPath();
+        let stdout = '';
+        let stderr = '';
 
-async function executeAzCliCommand(command: string, args?: string[]): Promise<string> {
-    let stdout = '';
-    let stderr = '';
-   
-    let code = await exec.exec(`"${azPath}" ${command}`, args, {
-        silent: true,
-        listeners: {
-            stdout: (data: Buffer) => {
-                stdout += data.toString();
-            },
-            stderr: (data: Buffer) => {
-              stderr += data.toString();
-            }
+        try {
+            core.debug(`"${azCliPath}" ${command}`);
+            await exec.exec(`"${azCliPath}" ${command}`, args, {

```

Comment:
does exec throw on error? I believe it returns a code which if not 0 is an error.

Reply:
it will throw if exit code is not zero. If you do not want it throw for non-zero exit code, you will have to set it in exec options.